### PR TITLE
Add Connext endpoints

### PR DESCRIPTION
**Pull request by kaushalrajbacancy**
## Description

- Added new endpoints: 
(1) `getSupportedAssets` : This returns supported tokens. 
(2) `getQuotes` : This fetches Quote Transactions. 
(3) `getTransactionStatus` : This fetches Transaction Status.

- **The default provider is Connext.**
## Types of changes
What types of changes does your code introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Further comments (optional)

- **It has been tested locally.**

image

### CODE REVIEW

File: src/sdk/data/data.module.ts [hunk 1]
```diff
@@ -249,4 +249,65 @@ export class DataModule {
       throw new Error(error.message || 'Failed to fetch exchange rates');
     }
   }
+
+  async getConnextSupportedAssets(chainId?: number): Promise<ConnextToken[]> {
```
Comment (ch4r10t33r) [inline src/sdk/data/data.module.ts:253]: I would have a single endpoint for any underlying provider. call it `getSupportedAssets` passing in an optional Provider (with default value as Connext).

File: src/sdk/data/data.module.ts [hunk 2]
```diff
@@ -249,4 +249,65 @@ export class DataModule {
...
+
+      const result = await this.apiService.makeRequest(API_ENDPOINTS.GET_CONNEXT_SUPPORTED_ASSETS, MethodTypes.GET, queryParams);
+
+      return result ? result.tokens : [];
+    } catch (error) {
+      throw new Error(error.message || 'Failed to get connext supported assets');
+    }
+  }
+
+  async getConnextQuotes(
```
Comment (ch4r10t33r) [inline src/sdk/data/data.module.ts:268]: Again, we ought to have a common endpoint called `getQuotes` with optional provider (set default as Connext).

File: src/sdk/data/data.module.ts [hunk 3]
```diff
@@ -249,4 +249,65 @@ export class DataModule {
...
+
+      const result = await this.apiService.makeRequest(API_ENDPOINTS.GET_CONNEXT_QUOTE_TRANSACTIONS, MethodTypes.GET, queryParams);
+
+      return result ? result.transactions : [];
+    } catch (error) {
+      throw new Error(error.message || 'Failed to get connext quotes transactions');
+    }
+  }
+
+  async getConnextTransactionStatus(fromChainId: number, toChainId: number, transactionHash: string): Promise<ConnextTransactionStatus> {
```
Comment (ch4r10t33r) [inline src/sdk/data/data.module.ts:297]: Pls see above.

File: src/sdk/data/data.module.ts [hunk 4]
```diff
@@ -249,4 +249,65 @@ export class DataModule {
...
+    try {
+      const queryParams = {
+        'api-key': this.currentApi,
```
Comment (ch4r10t33r) [inline src/sdk/data/data.module.ts:300]: Again. Why do we see this discrepancy everywhere? Can we stick to CamelCase throughout?

**Review by ch4r10t33r — CHANGES_REQUESTED** [review]

File: src/sdk/data/data.module.ts (hunk 4 above)
Comment (kaushalrajbacancy) [reply src/sdk/data/data.module.ts:300]: The `api-key` parameter is set by **TYK** to authenticate the user.

File: src/sdk/data/data.module.ts (hunk 1 above)
Comment (kaushalrajbacancy) [reply src/sdk/data/data.module.ts]: I have made this change.

File: src/sdk/data/data.module.ts (hunk 2 above)
Comment (kaushalrajbacancy) [reply src/sdk/data/data.module.ts]: I have made this change.

File: src/sdk/data/data.module.ts [hunk 5]
```diff
@@ -249,4 +250,96 @@ export class DataModule {
       throw new Error(error.message || 'Failed to fetch exchange rates');
     }
   }
+
+  async getSupportedAssets(chainId?: number, provider?: QuotesProvider): Promise<Token[]> {
```
Comment (ch4r10t33r) [inline src/sdk/data/data.module.ts:254]: `QuotesProvider` is an ambiguous name. What we are getting here are offers/orders, so we have call them accordingly. In this case, they are bridges, so we can name them BridgingProvider

File: src/sdk/dataUtils.ts [hunk 6]
```diff
@@ -224,4 +224,57 @@ export class DataUtils {
...
+
+    return this.dataModule.getSupportedAssets(chainId, provider);
+  }
+
+  /**
+  * gets quote transactions
+  * @param dto
+  * @return Promise<QuoteTransaction[]>
+  */
+  async getQuotes(dto: GetQuotesDto): Promise<QuoteTransaction[]> {
```
Comment (ch4r10t33r) [inline src/sdk/dataUtils.ts:244]: Again `QuotesTransaction` doesn't make sense. Pls call it Offers or just Quotes.

**Review by ch4r10t33r — CHANGES_REQUESTED** [review]

**Review by ch4r10t33r — APPROVED** [review]

**Review by lbw33 — APPROVED** [review]

**Review by 0xSulpiride — APPROVED** [review]

**Pull request merged**